bittide/src/Bittide/Counter.hs

```diff
@@ -0,0 +1,134 @@
+-- SPDX-FileCopyrightText: 2022 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+module Bittide.Counter
+  ( Active
+  , domainDiffCounter
+  ) where
+
+import Clash.Explicit.Prelude
+
+import Clash.Cores.Xilinx.Xpm (xpmCdcGray)
+import Clash.Sized.Extra (unsignedToSigned, glueUnsigneds)
+
+-- | State of 'domainDiffCounter'
+data DdcState
```

Comment:
This definition matches `EscState`, so maybe reduce it to a single one. This makes it also easier to notice the 64 bit size constraint, while this way a reader alwasy has to keep in mind that `DdcState` is hardcoded to be 64 bits.

Reply:
Other than that, its basically just `Maybe`. Decide on your own whether you like to keep use of that fact.